perfmap.go

```diff
@@ -220,6 +245,33 @@ func (m *PerfMap) BufferSize() int {
 	return m.bufferSize
 }
 
+// Telemetry returns the usage and lost telemetry
+func (m *PerfMap) Telemetry() (usage []uint64, lost []uint64) {
+	if m.usageTelemetry == nil || m.lostTelemetry == nil {
+		return nil, nil
+	}
+	usage = make([]uint64, len(m.usageTelemetry))
+	lost = make([]uint64, len(m.lostTelemetry))
+	for cpu := range m.usageTelemetry {
+		// reset to zero, so we return the max value between each collection
+		usage[cpu] = m.usageTelemetry[cpu].Swap(0)
+		lost[cpu] = m.lostTelemetry[cpu].Load()
+	}
+	return
+}
+
 func isPerfClosed(err error) bool {
 	return errors.Is(err, perf.ErrClosed)
 }
+
+func updateMaxTelemetry(a *atomic.Uint64, val uint64) {
+	for {
+		oldVal := a.Load()
+		if val <= oldVal {
+			return
+		}
+		if a.CompareAndSwap(oldVal, val) {
```

Comment:
Would ever return false here?

Reply:
Yes, the point is that if another goroutine managed to update the value underneath, you don't make a change.

Reply:
It shouldn't be possible based on the structure of other code, but this is the safe way to do it.

Reply:
So we would try again in that case?

Reply:
Yes, it would do the load, max check, and CAS again.